src/vaults/Vault.sol

```diff
@@ -736,6 +737,24 @@ abstract contract Vault is IVault, CoreReference, ReentrancyGuardUpgradeable, Va
         }
     }
 
+    function setDepositsEnabled(bool _enabled) external onlyStrategist whenPaused {
+        depositsEnabled = _enabled;
+    }
+
+    modifier onlyParticipantOrStrategist(address _user) {
+        (uint256 depositedToken0, uint256 pendingToken0, ) = this.token0Balance(_user);
+        (uint256 depositedToken1, uint256 pendingToken1, ) = this.token1Balance(_user);
+        require(
+            depositedToken0 + pendingToken0 + depositedToken1 + pendingToken1 > 0 || _isStrategist(_user),
+            "NOT_PARTICIPANT_OR_STRATEGIST"
+        );
+        _;
+    }
+    modifier whenDepositsEnabled() {
+        require(depositsEnabled, "DEPOSITS_DISABLED");
+        _;
+    }
```

Comment:
Let's put modifiers before function defs: https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html#layout-of-state-variables-in-storage

Reply:
I see this is the pattern in our files. Does this have an impact on storage layout? I don't see anything in the included link related to modifier/function ordering.

Reply:
Sorry, I used the wrong link - [this is the correct one](https://docs.soliditylang.org/en/v0.8.16/style-guide.html#order-of-layout). No impact, just a style guide